app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php

```diff
@@ -77,4 +77,70 @@ public function indexAction()
     {
         $this->_initAction()->renderLayout();
     }
+
+    /**
+     * Download requested log file
+     */
+    public function downloadAction()
+    {
+        $params = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        if (file_exists($fileName)) {
+            header('Content-Description: File Transfer');
+            header('Content-Type: application/octet-stream');
+            header('Expires: 0');
+            header('Cache-Control: must-revalidate');
+            header('Pragma: public');
+            header('Content-Disposition: attachment; filename="' . basename($fileName) . '"');
+
+            $file = file($fileName);
+            $lines = count($file);
+            if (is_array($file) && isset($params['rows']) && $lines > $params['rows']) {
+                $pointer = $lines - $params['rows'];
+                $text = "";
+                for (; $lines > $pointer; $pointer++) {
+                    $text .= $file[$pointer];
+                }
+                echo $text;
+                exit;
+            } else {
+                header('Content-Length: ' . filesize($fileName));
+                readfile($fileName);
+            }
+        } else {
```

Comment:
Avoid ```else``` in this case by moving this before the if with ```if (!file_exists($fileName)) {```.
Objectif reduce the imbrication